vmsize.go

```diff
@@ -0,0 +1,156 @@
+package skewer
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+// This file adds support for more capabilities based on VM naming conventions that includes vmsize parsing.
+// VM naming conventions are documented at: https://docs.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
+// Note: Some common capabilities like familyName and VCPUs, which can also be
+// fetched using the ResourceSKU API, are not included here. They can be found in sku.go.
+
+var skuSizeScheme = regexp.MustCompile(
+	`^([A-Z])([A-Z]?)([0-9]+)-?((?:[0-9]+)?)((?:[abcdilmtspPr]+|NP)?)_?((?:[A-Z][0-9]+)?)_?((?:[vV][1-9])?)(_Promo)?$`,
+)
+
+var partIndexes = map[string]int{
+	FamilyKey:           1,
+	SubFamilyKey:        2,
+	VCPUsKey:            3,
+	ConstrainedVCPUsKey: 4,
+	AdditiveFeaturesKey: 5,
+	AcceleratorTypeKey:  6,
+	VersionKey:          7,
+	PromoKey:            8,
+}
+
+// parseVMSize parses the VM size and returns the parts as a map.
+func parseVMSize(vmSizeName string) (map[string]string, error) {
+	if vmSizeName == "" {
+		return nil, fmt.Errorf("error parsing VM size, it is empty")
+	}
+
+	parts := skuSizeScheme.FindStringSubmatch(vmSizeName)
+
+	if parts == nil || len(parts) < 8 {
+		return nil, fmt.Errorf("could not parse VM size %s", vmSizeName)
+	}
+
+	result := make(map[string]string)
+	for part, index := range partIndexes {
+		result[part] = parts[index]
+	}
+	return result, nil
+}
+
+// GetSubFamily fetches the 'subFamily' part from the SKU size and returns it.
+func (s *SKU) GetSubFamily() (*string, error) {
+	return getSubFamily(s.GetSize())
```

Comment:
Can we parse only once (rather than on every property access), maybe on first use? And store the result on SKU? Or are we trying to optimize hard for memory use?